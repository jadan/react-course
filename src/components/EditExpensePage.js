import React from 'react';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const EditExpensePage = props => (
  <div>
    <button onClick={() => {
      props.dispatch(removeExpense({ id: props.expense.id }));
      props.history.push('/');
    }}
    >Remove
    </button>
    <ExpenseForm
      expense={props.expense}
      onSubmit={(expense) => {
        props.dispatch(editExpense(expense));
        props.history.push('/');
      }}
    />
  </div>
);

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

export default connect(mapStateToProps)(EditExpensePage);
