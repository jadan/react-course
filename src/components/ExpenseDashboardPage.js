import React from 'react';


const ExpenseDashboardPage = () => (
  <div>
    This is from my component.
  </div>
);

export default ExpenseDashboardPage;
