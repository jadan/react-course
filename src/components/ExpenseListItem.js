import moment from 'moment';
import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({
  description, amount, createdAt, id
}) => (
  // description, amount and createdAt
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p> {amount}- {moment(createdAt).format('MM-DD-YYYY')}</p>
  </div>
);

export default ExpenseListItem;
