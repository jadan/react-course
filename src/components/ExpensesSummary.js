import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
  const expenseWord = props.count > 1 ? ' expenses ' : ' expense ';
  const formattedTotal = numeral(props.expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      {props.expensesCount > 0 &&
      <h1>
        Viewing {props.expensesCount}
        {expenseWord}
        totalling {formattedTotal}
      </h1>}
    </div>);
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesTotal: selectExpensesTotal(visibleExpenses),
    expensesCount: visibleExpenses.length
  };
};

// HOC to add Redux to React
export default connect(mapStateToProps)(ExpensesSummary);
