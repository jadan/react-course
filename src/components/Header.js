import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ logout }) => (
  <div>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact>Go home</NavLink>
    <NavLink to="/create" activeClassName="is-active" exact>Create</NavLink>
    <button onClick={logout}>Logout</button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(startLogout())
});
export default connect(undefined, mapDispatchToProps)(Header);

