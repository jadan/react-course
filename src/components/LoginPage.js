import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ login }) => (
  <div>
    Please login.
    <button onClick={login}>Login</button>
  </div>
);

const mapDispatchToProps = dispatch => ({ login: () => dispatch(startLogin()) });

export default connect(undefined, mapDispatchToProps)(LoginPage);
