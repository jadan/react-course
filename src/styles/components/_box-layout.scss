.box-layout {
  background: url('/images/bg.jpg');
  background-size: cover;
  height: 100vh; // set to 100% of viewport height
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  border-radius: 3px;
  text-align: center;
  width: 25rem;
  padding: $xs-text $small-text;
  background: fade-out($color: #ffffff, $amount: 0.15);
}

.box-layout__title {
  margin: 0 0 $small-text 0;
  line-height: 1;
  font-size: $large-text;
}