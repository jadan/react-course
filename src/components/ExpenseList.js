import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

const ExpenseList = props => (
  <div>
    <h1>Expense List</h1>
    {props.expenses.map(expense => <ExpenseListItem key={expense.id} {...expense} />)}
  </div>
);

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters)
});


// HOC to add Redux to React
export default connect(mapStateToProps)(ExpenseList);

