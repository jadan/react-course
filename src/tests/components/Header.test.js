// react-test-renderer
import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

/* eslint-disable no-undef */
test('should render Header correctly', () => {
  const wrapper = shallow(<Header logout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
// should call logout on logout click
test('should call logout on logout click', () => {
  const onLogoutSpy = jest.fn();
  const wrapper = shallow(<Header logout={onLogoutSpy} />);
  wrapper.find('button').simulate('click');
  expect(onLogoutSpy).toHaveBeenCalled();
});

/* eslint-enable */
