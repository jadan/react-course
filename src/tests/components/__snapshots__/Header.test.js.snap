// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Go home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/create"
  >
    Create
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
